### ISSUE TITLE
livekit-client 1.14.0 broken in Angular app

### ISSUE THREAD

**Issue by pabloFuente** [livekit/client-sdk-js#893]
### What I'm expecting

Adding `livekit-client: 1.14.0` dependency to an Angular 16 project should work fine.

### What happens instead

Compilation of Angular 16 app (`ng build`) fails with the following error

```
Error: node_modules/livekit-client/dist/src/room/track/LocalAudioTrack.d.ts:24:51 - error TS2304: Cannot find name 'this'.

24     setProcessor(processor: TrackProcessor<typeof this.kind>): Promise<void>;
                                                     ~~~~
```

Which seems related to the [track-processors API](https://github.com/livekit/track-processors-js).

### Reproduction

The error is easily replicable with the following steps:

```bash
# Create new Angular 16 project
ng new test-project --defaults

# Install livekit-client dependency
cd test-project && npm i livekit-client@1.14.0

# Add an import of the livekit-client in the project
sed -i '1s/^/import * as LiveKit from "livekit-client";\n/' src/app/app.component.ts

# Try to build. THIS FAILS
ng build
```

### Logs

```shell
user@user-pro:~/Desktop/test-project$ ng build
✔ Browser application bundle generation complete.

Error: node_modules/livekit-client/dist/src/room/track/LocalAudioTrack.d.ts:24:51 - error TS2304: Cannot find name 'this'.

24     setProcessor(processor: TrackProcessor<typeof this.kind>): Promise<void>;
                                                     ~~~~
```

### System Info

```shell
Versions:

- OS: linux x64 (Ubuntu 22.04.3 LTS)
- Node: 18.18.2
- Package Manager: npm 9.8.1
- Angular: 16.2.9
- livekit-client: 1.14.0
```

### Severity

blocking all usage of LiveKit

### Additional Information

There is a way of fixing the issue manually, which is making an import in file `node_modules/livekit-client/dist/src/room/track/LocalAudioTrack.d.ts`:

```typescript
import *  as TrackKind from './Track';
```

And replacing `<typeof this.kind>` with `<TrackKind.Track.Kind>` in line node_modules/livekit-client/dist/src/room/track/LocalAudioTrack.d.ts:24:51

But of course this is far from ideal.

**Comment by lukasIO** [comment]
Hi, 

could you post the exact error message you get? It can't be the same as in #166 because we have since replaced the protobuf library that imported those paths. 
Are you seeing "only" the logger error at first?

**Comment by pabloFuente** [comment]
The complete log for the first error is the following:

```
user@user-pro:~/Documents/Git/test-project$ ng serve
✔ Browser application bundle generation complete.

Initial Chunk Files   | Names         |  Raw Size
vendor.js             | vendor        |   5.07 MB | 
polyfills.js          | polyfills     | 514.76 kB | 
styles.css, styles.js | styles        | 309.57 kB | 
main.js               | main          | 136.82 kB | 
runtime.js            | runtime       |   6.54 kB | 

                      | Initial Total |   6.02 MB

Build at: 2023-10-17T16:24:12.821Z - Hash: fe55f98ebaed05df - Time: 10399ms


Error: node_modules/livekit-server-sdk/dist/proto/livekit_egress.d.ts:1:8 - error TS1192: Module '"/home/user-pro/Documents/Git/test-project/node_modules/protobufjs/minimal"' has no default export.

1 import _m0 from "protobufjs/minimal";
         ~~~


Error: node_modules/livekit-server-sdk/dist/proto/livekit_ingress.d.ts:1:8 - error TS1192: Module '"/home/user-pro/Documents/Git/test-project/node_modules/protobufjs/minimal"' has no default export.

1 import _m0 from "protobufjs/minimal";
         ~~~


Error: node_modules/livekit-server-sdk/dist/proto/livekit_models.d.ts:1:8 - error TS1192: Module '"/home/user-pro/Documents/Git/test-project/node_modules/protobufjs/minimal"' has no default export.

1 import _m0 from "protobufjs/minimal";
         ~~~


Error: node_modules/livekit-server-sdk/dist/proto/livekit_room.d.ts:1:8 - error TS1192: Module '"/home/user-pro/Documents/Git/test-project/node_modules/protobufjs/minimal"' has no default export.

1 import _m0 from "protobufjs/minimal";
         ~~~


Error: node_modules/livekit-server-sdk/dist/proto/livekit_webhook.d.ts:1:8 - error TS1192: Module '"/home/user-pro/Documents/Git/test-project/node_modules/protobufjs/minimal"' has no default export.

1 import _m0 from "protobufjs/minimal";
         ~~~




** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **


✖ Failed to compile.

```

But again, I know how to get rid of this error. Is the second one the one that has no solution for now.

**Comment by lukasIO** [comment]
had a closer look at this. Solving it differently without breaking backwards compatibility seems tricky. 
I found this thread talking about the whole thing, one suggestion is to use `TrackProcessor<this['kind']>` instead of the `typeof this.kind` approach. Could you give that a try?

**Comment by lukasIO** [comment]
> Well, that solution is pretty much the same as I described in my Additional Information section in my first comment:

It's not the same, as it will keep the actual type information. `TrackKind.Track.Kind` as you suggested doesn't specify whether the track is an audio or a video track. 

Did you confirm that `TrackProcessor<this['kind']>` works for your Angular setup? If that's the case then I can open a PR with that change and change it in the client sdk directly.

**Comment by pabloFuente** [comment]
I hace confirmed that replacing `TrackProcessor<typeof this.kind>` with `TrackProcessor<this["kind"]>` in `node_modules/livekit-client/dist/src/room/track/LocalAudioTrack.d.ts:24:51` fixes the TypeScript compilation problem.
I am guessing that replacing [**these occurrences**](https://github.com/search?q=repo%3Alivekit%2Fclient-sdk-js+%22typeof+this.kind%22&type=code) with `this['kind']` should be enough to bypass this TypeScript "limitation".

**Comment by lukasIO** [comment]
nice, thanks! would you want to open a PR for it?

**Comment by pabloFuente** [comment]
Sure. I've already opened [this one](https://github.com/livekit/client-sdk-js/pull/899) for a typo fix in the documentation, so I'm familiar with the process :smiley:

**Comment by lukasIO** [comment]
closed via #901
